src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Publish.targets

```diff
@@ -826,6 +861,15 @@ Copyright (c) .NET Foundation. All rights reserved.
                                             Condition="'%(_NoneWithTargetPath.CopyToPublishDirectory)'=='PreserveNewest'"/>
     </ItemGroup>
 
+    <ItemGroup Condition="'$(PublishSingleFile)' == 'true' and '$(SelfContained)' == 'true' and '$(_TargetFrameworkVersionWithoutV)' >= '5.0'">
```

Comment:
> Would it make sense to condition this on the existence of the SingleFileHostIntermediatePath instead? Basically rely on whatever logic is in the _CreateSingleFileHost.

I think it'd have to be in addition to, otherwise it could grab the intermediate left over from a previous single-file publish even if the current publish isn't.

> Also - what happens if _CreateSingleFileHost doesn't produce a file (for example if SingleFileHostSourcePath is empty) - should we fail here instead?

Maybe `_CreateSingleFileHost` should error out instead - so condition the target on if we should be using singlefilehost and error if `SingleFileHostSourcePath` is empty or doesn't exist rather than having that be part of the target's condition.